main.go

```diff
@@ -42,6 +52,69 @@ type Output struct {
 	Expiration      string
 }
 
+type Cache struct {
+	Dir string
+}
+
+func NewCache(dir string) (*Cache, error) {
+	if _, err := os.Open(dir); err != nil {
+		return nil, err
+	}
+
+	fn := path.Join(dir, "validate")
+	if _, err := os.Create(fn); err != nil {
+		return nil, err
+	}
+	os.Remove(fn)
+
+	return &Cache{
+		Dir: dir,
+	}, nil
+}
+
+func (c *Cache) filenameByRole(role string) string {
+	h := sha256.New()
+	h.Write([]byte(role))
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+func (c *Cache) Get(role string) (*Output, bool) {
+	fn := path.Join(c.Dir, c.filenameByRole(role))
+	if f, err := os.Open(fn); err != nil {
+		return nil, false
+	} else {
+		if bytes, err := io.ReadAll(f); err != nil {
+			return nil, false
+		} else {
```

Comment:
unnecessary else

Reply:
I need the `bytes` in the scope so I can't get rid of else.  I can refactor it to unroll the scope but I prefer to keep the scope limited.